File: 1605.py
class Solution:
    def restoreMatrix(self, rowSum: list[int], colSum: list[int]) -> list[list[int]]:
        n, m = len(rowSum), len(colSum)
        matrix = [[0] * m for _ in range(n)]
        for i in range(n):
            for j in range(m):
                matrix[i][j] = min(rowSum[i], colSum[j])
                rowSum[i] -= matrix[i][j]
                colSum[j] -= matrix[i][j]
        return matrix
        
if __name__ == '__main__':
    s = Solution()
    rowSum = list(map(int, input().split()))
    colSum = list(map(int, input().split()))
    print(s.restoreMatrix(rowSum, colSum))

File: cpp/1605.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    vector<vector<int>> restoreMatrix(vector<int>& rowSum, vector<int>& colSum) {
        int n = rowSum.size(), m = colSum.size();
        vector<vector<int>> matrix(n, vector<int>(m, 0));
        for(int i = 0; i < n; i++){
            for(int j = 0; j < m; j++){
                matrix[i][j] = min(rowSum[i], colSum[j]);
                rowSum[i] -= matrix[i][j];
                colSum[j] -= matrix[i][j];
            }
        }
        return matrix;
    }
};

int main(){
    int n, m;
    cin >> n >> m;
    vector<int> rowSum(n), colSum(m);
    for(int i = 0; i < n; i++){
        cin >> rowSum[i];
    }
    for(int i = 0; i < m; i++){
        cin >> colSum[i];
    }
    Solution s;
    vector<vector<int>> res = s.restoreMatrix(rowSum, colSum);
}